lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -157,14 +157,16 @@ defmodule ExUnit.Assertions do
       quote do
         value = unquote(value)
 
-        unless value do
-          raise ExUnit.AssertionError,
-            args: unquote(args),
-            expr: unquote(escape_quoted(:assert, [], assertion)),
-            message: "Expected truthy, got #{inspect(value)}"
-        end
+        case truthy?(value) do
```

Comment:
If we convert this `unless` into a case and call suppress warnings on it, would it be enough?